core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks.go

```diff
@@ -0,0 +1,828 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package main
+
+import (
+    "fmt"
+    "errors"
+    "time"
+    "strconv"
+    "encoding/json"
+
+    "github.com/golang/protobuf/proto"
+    "github.com/hyperledger/fabric-chaincode-go/shim"
+    log "github.com/sirupsen/logrus"
+    "github.com/hyperledger-labs/weaver-dlt-interoperability/core/network/fabric-interop-cc/interfaces/asset-mgmt/protos-go/common"
+)
+
+
+const(
+    defaultTimeLockMillis = 5 * 60 * 1000      // 5 minutes
+)
+
+
+// AssetManagement provides functions to manage asset locks by calling suitable functions in the interop contract
+type AssetManagement struct {
+    shim.Chaincode
+    interopChaincodeId string
+}
+
+
+// Utility functions
+func (am *AssetManagement) Configure(interopChaincodeId string) {
```

Comment:
Do we assume that access controls around who would be allowed to call this function are controlled by the application? If not, I assume this introduces a point of vulnerability if not controlled? Is there a strong reason this should be dynamic as opposed to set once during initialization?

Reply:
This code is not going to run in a separate module. It's part of any application chaincode that wishes to inherit and use it. So there is no way to initialize its state (i.e., the interop chaincode ID) other than through the application chaincode (typically in the init function).